---
title: "Bios 8060E Project"
author: "Amanda Blubaugh"
date: 'last updated: 11/2/2019'
output:
  html_document
---

# Project Overview

### Transcriptomic profile differences in atopic dermatitis (AD) canid house dust mite (HDM) models when considering using multifactorial differential expression for sensitization of model dogs

Perform differential expression (DE) of previous data from an AD canine model paper in both a multifactoral DE that mimics the original paper (using R open source versus proprietary software of the paper) and in a classic one factor DE without considering an the additional factor of previous canine sensitization to the HDM model. I would like to see how immunological genes that are significantly up or down-regulated change between the two DE evaluations.

This data set has 12 total canine samples: 6 dogs have been sensitized to the House Dust Mite (HDM) and 6 have not.  All dogs are exposed to the HDM and sampled at a 24 hour point after exposure through a patch with HDM crushed paste or a mineral oil patch control. All dogs were also sampled at a 0h, healthy timepoint with no exposure to anything.

The original paper provides a multifactorial differential expression of genes in HDM samples versus 0h control healthy, as well as versus 24h saline, allowing for sensitization of the HDM or not as a factor. 

Traditional atopic dermatitis (exzema) canine models for development of human and canine drugs to treat the disease require that all dogs be sensitized to HDM prior to recieving HDM stimulus for modelled skin to AD. canine AD models are essential for drug development in treating AD for humans and canines. 

## Goals

Since this paper contains 6 subjects with no previous sensitization, and 6 subjects with sensitization, the goal of this project will be to:

1. Replicate the multifactoral DE of data for all 12 dogs (sensitized or not being the additional factor) at 24h HDM versus 0h untreated no exposure.

2. Do additional individual DE of 6 vs. 6 dogs with no multifactoral consideration for sensitization, since the larger group is divided into two smaller groups that dilineate this factor.

## Hypotheses
It is hypothesized that:

1. Replication of the orginal results will be possible for confirmation of proper manipulation of the data in this study on a multifactoral level and reproducibility to validate the study results.

2. Division of the sample group into smaller, even groups of dogs that have been sensitized or not will provide a DE view that shows differences in DE based on previous sensitization or not, validating whether or not sensitization is required in a HDM canine atopic model.

## Limitations and Justifications

The sample size is quite low for this study, but this is some of the only sample data available for canine AD models to compare sensitized or non- sensitized dogs.  We are also not utilizing saline controls of this study, which are still controversial in whether saline itself produces an inflammatory response to the skin.

There is no previous study evaluating the efficacy of first sensitizing dogs to HDM prior to inducing AD models with HDM paste.

## Project Data Source

Samples from a previously published paper by [Schamber et al. (2014)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4199687/) will be used for this project and is in project resources.

# Materials and Methods

## Downloading Data from GEOquery (NCBI GEO database)

Prior to working in R studio, I installed [Bioconductor](https://www.bioconductor.org/install/) resources on my local computer through the console, as well as [GEOquery](https://bioconductor.org/packages/release/bioc/html/GEOquery.html) and [limma](https://bioconductor.org/packages/release/bioc/html/limma.html)
<br>
In the console:
```
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install()


BiocManager::install(c("Biobase","GEOquery","limma"))

```
<br>

## Load libraries for use
```{r load_libraries}
library(limma)
library(Biobase)
library(GEOquery)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
```

## Cache data from GEO datasets
```{r get_data}

#You can search in the NCBI GEOquery database with the GSE number: GSE58442
#The getGEO function in GEOquery library uses this identifier to gather our data
gset <- getGEO("GSE58442", GSEMatrix =TRUE, AnnotGPL=FALSE)
if (length(gset) > 1) idx <- grep("GPL18789", attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]
```

### Inspect Uncleaned Data
```{r inspection}
head(gset)
head(pData(gset))
```

## Preliminary Analysis with GEO2R
```{r GEO2R_preliminary}
# make proper column names to match toptable 
fvarLabels(gset) <- make.names(fvarLabels(gset))
fvarLabels(gset)

# group names for all samples
gsms <- "013240132401324013240132401324013240132401324013240132401324"
sml <- c()
for (i in 1:nchar(gsms)) { sml[i] <- substr(gsms,i,i) }

# log2 transform
ex <- exprs(gset)
qx <- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
LogC <- (qx[5] > 100) ||
  (qx[6]-qx[1] > 50 && qx[2] > 0) ||
  (qx[2] > 0 && qx[2] < 1 && qx[4] > 1 && qx[4] < 2)
if (LogC) { ex[which(ex <= 0)] <- NaN
exprs(gset) <- log2(ex) }

# set up the data and proceed with analysis
sml <- paste("G", sml, sep="")    # set group names
fl <- as.factor(sml)
fl
gset$description <- fl

# Create and view design matrix
design <- model.matrix(~ description + 0, gset)

colnames(design) <- levels(fl)
head(design)

# lm function fit with single variable
fit <- lmFit(gset, design)
fit2 <- eBayes(fit)
topTable(fit2)

```
In this preliminary analysis, we can see which probes had the lowest/best scoring False Discovery Rate (FDR)/ adjusted p-value for change in probe fluoresence (number of bound, tagged fluorescent transcripts to the probe on the microarray) with a basic linear model, adjusted with a Bayesian fit. I don't have probes that have a gene symbol attached to them separated from those without a gene symbol. I also haven't viewed other aspects of the data yet (principal component analysis (PCA) for example).


## Aquire additional phenotypic data for local use of samples in this study
Going to [GEO2R](https://www.ncbi.nlm.nih.gov/geo/geo2r/), we can obtain the phenotypic data attached to each sample and label them by number for the design above. This is a way to manipulate the data and get preliminary results without having the raw data directly on the local computer being used.

Phenotypic data for each raw .txt file can be highlighted and read into a .txt file for manipulation with the raw data (i.e. to clean it up in a way we can acheive ultimate our goals in the study)

```{r load_metadata}
#Here, we locate where we saved the .txt file we got for reading metadata into a table, notating that the file has tabular separations ("\t" for sep), that there is a header for each column (our variables; header=TRUE), and that we have row names which are the same .txt GSM# .txt files that we have read in previously.

META <- read.AnnotatedDataFrame("./data/processed_data/Schamber_phenotypes.txt", sep="\t", header=TRUE, row.names=1)
head(pData(META))


```

In this metadata information table I have created, I have manually added color identifications in that were not part of the phenotyptic table copied to a .txt file from GEO2R. This is so when I run PCA, we can see which groups may naturally diverge. 

## Principal Component Analysis
In PCA, the heirarchical clustering algorithm does not take phenotypic groups into account. Using our additional metadata coloring, we can create a PCA with the microarry data we have.

```{r pca}
pca <- prcomp(scale(t(exprs(gset))))
summary(pca)

colors <- META$ColorSens
labels <- META$Label
legendlabels <- c("0h no Tx","0h no Tx sensitized","6h saline", "6h saline sensitized","24h saline","24h saline sensitized","6h HDM","6h HDM sensitized","24h HDM","24h HDM sensitized")
legendcolors <- c("darkgreen","darkolivegreen1","darkorchid4","orchid2","darkblue","royalblue1","gold","lightgoldenrod2","firebrick","coral1")

plot(pca$x[,1:2],  col=colors, pch=19, cex=2.0)
text(pca$x[,1:2], labels=labels, col="black", pos=1, cex=0.5)
legend("topleft", legend= legendlabels, col=legendcolors, bty="n", horiz=FALSE, cex=0.75, pch=19)

#this pca was finalized with the same code above after the line tiff("Schamber_2dPCA_preliminary.tiff", width=10, height=10, units='in', res=300)
knitr::include_graphics("./results/Schamber_2d_PCA_with_all_variables_lower_res.png")

```

## Back to data for additional cleaning

```{r get_gene_symbols}
head(pData(gset))

#head(exprs(gset))

annotation <- read.csv("./data/raw_data/GPL18789-28160.txt", sep="\t", header=TRUE)

probe_to_gene <- annotation[,c(1,10)]
merged_data <- merge(probe_to_gene, exprs(gset), by="row.names")
#merged_data

```

```{r cleaned_up}

#Remember, our exprs(gset) already had the variance stabilized accross the mean
merged_data <- as.data.frame(merged_data)
#merged_data

no_gene_known_removed <- merged_data %>%
  filter(merged_data$GENE_SYMBOL != "")

numbers_gone <- no_gene_known_removed %>%
  select(-c(Row.names, ID))

#numbers_gone

#design

#Did the code below to make the final raw data copy for data to be used in this experiment
#write.table(numbers_gone, "./data/raw_data/merged_raw_data.txt", sep="\t")


```

## Set Up Design Tables

G0 = 0h untreated
G1 = 6h HDM
G2 = 6h saline
G3 = 24h HDM
G4 = 24h saline

Further, half of the dogs are sensitized and half are not, which is also info in our metadata table.

```{r design_groups}

#Looking back at our initial results from our exploratory analysis

#topTable(fit2)

```


# Resources

## References


[Schamber et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4199687/) Gene Expression in the Skin of Dogs Sensitized to the House Dust Mite (_Dermatophagoides farinae_). G3: Genes, Genomics, Genetics. 2014 Oct; 4(10): 1787-1795. [Accession #GSE58442](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE58442)

Edgar R, Domrachev M, Lash AE.
Gene Expression Omnibus: NCBI gene expression and hybridization array data repository
Nucleic Acids Res. 2002 Jan 1;30(1):207-10
[GEO (Gene Expression Ombnibus) and GEO2R](https://www.ncbi.nlm.nih.gov/geo/)

[Simon et al.](https://brb.nci.nih.gov/techreport/DesignandAnalysisofDNAMicroarrayInvestigations.pdf) Design and Analysis of DNA Microarray Investigations. Springer-Veriag, New York. ISBN ISBN 0-387-00135-2. pp 66-69: Differential Gene Expression in Microarrays.

## R Packages
```{r citations}
citation("Biobase")
citation("GEOquery")
citation("limma")

```


## Sample Source Information of raw data from [GEO Data Sets](https://www.ncbi.nlm.nih.gov/gds)

1. N_2847_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411196/
Sample		Accession: GSM1411196	ID: 301411196

2. N_2847_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411195/
Sample		Accession: GSM1411195	ID: 301411195

3. N_2845_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411191/
Sample		Accession: GSM1411191	ID: 301411191

4. N_2845_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411190/
Sample		Accession: GSM1411190	ID: 301411190

5. N_2845_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411187/
Sample		Accession: GSM1411187	ID: 301411187

6. N_2802_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411186/
Sample		Accession: GSM1411186	ID: 301411186

7. N_2802_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411185/
Sample		Accession: GSM1411185	ID: 301411185

8. N_2802_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411182/
Sample		Accession: GSM1411182	ID: 301411182

9. N_2800_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411181/
Sample		Accession: GSM1411181	ID: 301411181

10. N_2800_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411180/
Sample		Accession: GSM1411180	ID: 301411180

11. N_2847_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411192/
Sample		Accession: GSM1411192	ID: 301411192

12. N_2800_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411177/
Sample		Accession: GSM1411177	ID: 301411177

13. N_2716_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411176/
Sample		Accession: GSM1411176	ID: 301411176

14. N_2716_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411175/
Sample		Accession: GSM1411175	ID: 301411175

15. N_2716_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411172/
Sample		Accession: GSM1411172	ID: 301411172

16. N_2704_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411171/
Sample		Accession: GSM1411171	ID: 301411171

17. N_2704_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411170/
Sample		Accession: GSM1411170	ID: 301411170

18. N_2704_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411167/
Sample		Accession: GSM1411167	ID: 301411167

19. A_2855_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411166/
Sample		Accession: GSM1411166	ID: 301411166

20. A_2855_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411165/
Sample		Accession: GSM1411165	ID: 301411165

21. A_2855_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411162/
Sample		Accession: GSM1411162	ID: 301411162

22. A_2853_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411161/
Sample		Accession: GSM1411161	ID: 301411161

23. A_2853_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411160/
Sample		Accession: GSM1411160	ID: 301411160

24. A_2853_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411157/
Sample		Accession: GSM1411157	ID: 301411157

25. A_2841_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411156/
Sample		Accession: GSM1411156	ID: 301411156

26. A_2841_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411155/
Sample		Accession: GSM1411155	ID: 301411155

27. A_2841_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411152/
Sample		Accession: GSM1411152	ID: 301411152

28. A_2840_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411151/
Sample		Accession: GSM1411151	ID: 301411151

29. A_2840_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411150/
Sample		Accession: GSM1411150	ID: 301411150

30. A_2840_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411147/
Sample		Accession: GSM1411147	ID: 301411147

31. A_2839_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411146/
Sample		Accession: GSM1411146	ID: 301411146

32. A_2839_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411145/
Sample		Accession: GSM1411145	ID: 301411145

33. A_2839_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411142/
Sample		Accession: GSM1411142	ID: 301411142

34. A_2838_24h_S
Organism:	Canis lupus familiaris
Source name:	skin_treated_saline_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411141/
Sample		Accession: GSM1411141	ID: 301411141

35. A_2838_24h_A
Organism:	Canis lupus familiaris
Source name:	skin_treated_allergen_24h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411140/
Sample		Accession: GSM1411140	ID: 301411140

36. A_2838_0h
Organism:	Canis lupus familiaris
Source name:	skin_untreated_0h
Platform: GPL18789 Series: GSE58442 
FTP download: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1411nnn/GSM1411137/
Sample		Accession: GSM1411137	ID: 301411137